### ISSUE TITLE
Compiler invoked with (swank:init), fails for building Allegro runtimes sans compiler

### ISSUE THREAD

**Issue by genworks** [slime/slime#358]
Between 2.14 and 2.17, a `with-compilation-unit` was added in line 265 of `swank-loader.lisp`. 

On Allegro, this has the effect of invoking the compiler, whether any files actually need to be compiled or not. The result is that when trying to build swank into an Allegro CL runtime, a bus error is signaled, because by default in an Allegro runtime, the compiler is not available in the Lisp. 

Previously I was able to build Swank into Allegro runtimes simply by ensuring that all the swank files were compiled ahead of time, such that the fasls were up to date, and then compiler would not be invoked during the runtime build. 

I am not sure of the best fix here. A possible fix could be: 

```
(defun load-swank (&key (src-dir *source-directory*)
                     (fasl-dir *fasl-directory*)
                        quiet)
  (#-(and allegro (not compiler)) with-compilation-unit 
   #+(and allegro (not compiler)) progn ()
    (compile-files (src-files *swank-files* src-dir) fasl-dir t quiet))
  (funcall (q "swank::before-init")
           (slime-version-string)
           (list (contrib-dir fasl-dir)
                 (contrib-dir src-dir))))
```

This works for me now, but only addresses Allegro of course. There may be other CLs (LW?) which have similar problems when generating runtimes. 

Any better ideas out there, or would the above workaround be considered for merging into Slime in the meantime?

**Comment by luismbo** [comment]
1615c0aaf9f905f32a0b09a2d3b0a6f3554740c5 was it. @stassats any ideas?

**Comment by stassats** [comment]
Dunno. Just remove with-compilation-unit?

**Comment by genworks** [comment]
Well, would there be any downside to removing the `with-compilation-unit`? 

I assume it was put there for a reason. 

If no apparent downside to removing it then yes, that would be helpful for being able to include swank in Allegro runtime applications.  Otherwise the conditional proposed above will remove the `with-compilation-unit` only for the case of Allegro runtime.

**Comment by stassats** [comment]
It avoid warnings about undefined stuff.

**Comment by genworks** [comment]
Right, with-compilation-unit makes sense here for the normal use case. That's why it seems useful to leave the with-compilation-unit and just conditionalize it out for the case of Allegro & Runtime.

**Comment by stassats** [comment]
Can you come up with a pull request?
